### PR TITLE
Remove SwiftASTContext from Materializer.cpp (NFC)

### PR DESCRIPTION
**Pull request by adrian-prantl**

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

**Review by augusto2112 — APPROVED** [review]

**Comment by adrian-prantl** [comment]
@swift-ci test linux

**Comment by adrian-prantl** [comment]
@swift-ci test windows

**Comment by rintaro** [comment]
@swift-ci please test Linux

File: lldb/source/Expression/Materializer.cpp [hunk 1]
```diff
@@ -662,12 +657,6 @@ class EntityVariableBase : public Materializer::Entity {
 
       CompilerType valobj_type = valobj_sp->GetCompilerType();
 
-#ifdef LLDB_ENABLE_SWIFT
-      if (SwiftASTContext::IsGenericType(valobj_type)) {
-        valobj_sp = valobj_sp->GetDynamicValue(lldb::eDynamicDontRunTarget);
-      }
-#endif // LLDB_ENABLE_SWIFT
```
Comment (kastiglione) [inline lldb/source/Expression/Materializer.cpp:669]: why is this no longer needed?

Comment (adrian-prantl) [reply lldb/source/Expression/Materializer.cpp:669]: It was very old code that had no longer any effect on the test suite.

**Pull request merged**